### PR TITLE
Committed service for exhibits

### PR DESCRIPTION
**Pull request by ctxdl**
Service for exhibits

### CODE REVIEW

File: app/mobile-content/exhibits/exhibits.component.ts [hunk 1]
```diff
@@ -5,6 +5,12 @@ import { CreateExhibitDialogComponent } from './create-exhibit-dialog/create-exh
 import { DeleteExhibitDialogComponent } from './delete-exhibit-dialog/delete-exhibit-dialog.component';
 import { Exhibit } from './shared/exhibit.model';
 import { Status, statusType } from '../shared/status.model';
+import { ExhibitService } from './shared/exhibit.service';
+
+
+import { ToasterService } from 'angular2-toaster';
```
Comment (patrickrobrecht) [inline app/mobile-content/exhibits/exhibits.component.ts]: CG: First imports from external package, ordered by package name; empty line; imports from our own code ordered alphabetically by name

File: app/mobile-content/exhibits/exhibits.component.ts [hunk 2]
```diff
@@ -16,73 +22,124 @@ export class ExhibitsComponent implements OnInit {
...
-  ngOnInit() {
-    // TODO: replace dummy data with appropriate API calls
-    this.exhibits = new Array(30);
-    this.totalItems = this.exhibits.length;
-    for (let i = 0; i < this.exhibits.length; i++) {
-      this.exhibits[i] = Exhibit.getRandom();
+    ngOnInit() {
+      this.getPage(1);
     }
-    this.routes = ['ALL', 'Route 1', 'Route 2', 'Route 3', 'Route 4'];
```
Comment (patrickrobrecht) [inline app/mobile-content/exhibits/exhibits.component.ts]: Please leave at least the default option (all routes) here. The list of routes can be added after the route service is merged.

File: app/mobile-content/exhibits/exhibits.component.ts [hunk 3]
```diff
@@ -16,73 +22,124 @@ export class ExhibitsComponent implements OnInit {
...
 
   resetSearch() {
+    this.searchQuery = '';
+    this.exhibits = undefined;
+    this.exhibitCache.clear();
+    this.getPage(1);
     this.showingSearchResults = false;
-    // TODO: implement search reset
+  }
+  private translate(data: string): string {
```
Comment (patrickrobrecht) [inline app/mobile-content/exhibits/exhibits.component.ts]: Missing line between methods

File: app/mobile-content/exhibits/shared/exhibit-page.model.ts [hunk 4]
```diff
@@ -11,11 +11,11 @@ export class ExhibitPage {
...
   constructor(public text: string,
-              public exhibitId: number,
-              public status: statusType,
-              public type: pageType,
-              public audio: number,
-              public image: number,
-              public images: sliderImage[],
-              public hideYearNumbers: boolean) {}
-}
+    public exhibitId: number,
```
Comment (patrickrobrecht) [inline app/mobile-content/exhibits/shared/exhibit-page.model.ts]: Code style was fine before. Why did you make changes?

File: app/mobile-content/exhibits/shared/exhibit.model.ts [hunk 5]
```diff
@@ -1,40 +1,83 @@
...
+      return exhibit;
+    }
+    // Sorts the Exhibits by name
+    public static exhibitAlphaCompare(a: Exhibit, b: Exhibit): number {
+      return a.name.localeCompare(b.name);
+    }
+
+    // Returns a random dummy exhibit
+    // TODO: remove this method after server side is ready
+    static getRandom() {
```
Comment (patrickrobrecht) [inline app/mobile-content/exhibits/shared/exhibit.model.ts]: `getRandom()` should not be used anymore -> remove

File: app/mobile-content/exhibits/shared/exhibit.service.ts [hunk 6]
```diff
@@ -0,0 +1,126 @@
+import { Injectable } from '@angular/core';
+import { Response } from '@angular/http';
```
Comment (patrickrobrecht) [inline app/mobile-content/exhibits/shared/exhibit.service.ts]: order of imports, indentation of whole file

File: app/mobile-content/exhibits/shared/exhibit.service.ts [hunk 7]
```diff
@@ -0,0 +1,126 @@
...
+    }
+    /**
+     * Retrieves a subset of all exhibits based on supplied filter parameters.
+     * Returns an object with two keys:
+     * `items` an array of Topic objects that satisfy supplied search parameters and
+     * `metadata` an object containing info on the returned subset (page number, total results, etc.)
+     * @param page Page number for pagination.
+     * @param pageSize Amount of users per page.
+     * @param query Additional query to look for in topic title and description.∫
+     * @param status Only return exhibits with specified status.
```
Comment (patrickrobrecht) [inline app/mobile-content/exhibits/shared/exhibit.service.ts]: orderBy parameter missing in documentation

File: app/mobile-content/exhibits/shared/exhibit.service.ts [hunk 8]
```diff
@@ -0,0 +1,126 @@
...
+                    return {
+                        items: Exhibit.extractPaginatedArrayData(response),
+                        total: response.json().total
+                    };
+                }
+            ).catch(
+                (error: any) => this.handleError(error)
+            );
+    }
+    getTagNames(ids: any): Promise<any> {
```
Comment (patrickrobrecht) [inline app/mobile-content/exhibits/shared/exhibit.service.ts]: `getTagNames` should be moved to the tag service before a merge.

File: app/mobile-content/exhibits/shared/exhibit.service.ts (hunk 6 above)
Comment (ctxdl) [inline app/mobile-content/exhibits/shared/exhibit.service.ts]: The indentation has been fixed the order of the imputs in this fine seems to be ok? or am i missing something

**Comment by ctxdl** [comment]
The function getRandom is used by the routes component for dummy data when the route service is commited the function will be deleted

File: app/mobile-content/exhibits/shared/exhibit.service.ts (hunk 6 above)
Comment (patrickrobrecht) [inline app/mobile-content/exhibits/shared/exhibit.service.ts]: It should be:
`import { Injectable } from '@angular/core';
import { Response } from '@angular/http';
import { BehaviorSubject } from 'rxjs/Rx';`
(our CG: external imports ordered by package name of the package imported from due to the fact that multiple classes can be imported from the same package)

**Pull request merged**